### ISSUE TITLE
Discover shows videos in languages other than the selected ones

### ISSUE THREAD

**Issue by alex9099** [Chocobozzz/PeerTube#3152]
**What happened?**
Discover tab shows all videos, independent of the language

**What do you expect to happen instead?**
Show only videos with the configured languages

**Steps to reproduce:**

1.Choose languages on "Only display videos in the following languages/subtitles"

2.Click on discover

3.All videos are shown

**Additional information**
* PeerTube version or URL:2.4.0
* Browser name/version: Firefox

**Comment by Alterak** [comment]
I support the author! It is very uncomfortable! But most likely this is because in such videos, the Russian language was selected when loading ... maybe I'm wrong.

**Comment by alex9099** [comment]
>the Russian language was selected when loading 

What do you mean by that? I have portuguese and english selected and french, spanish and even arabic videos show up

**Comment by Alterak** [comment]
> What do you mean by that? I have portuguese and english selected and french, spanish and even arabic videos show up

Then this is a mistake ... in my section of similar videos are also in different languages.

**Comment by alex9099** [comment]
Yes, seems like the "following" sugested video also doesn't respect the language option (either unknown or some other languages are shown)

**Comment by merlin1608** [comment]
Hi, This feature should be must, many users are discouraged when they "discover" videos with mostly languages they don't understand.
Thanks in advance.
